docs/syntax/gantt.md

```diff
@@ -478,6 +478,21 @@ To hide the marker, set `todayMarker` to `off`.
 todayMarker off
 ```
 
+## Working hours and durations
+
+You can assign core working hours within the Gantt by providing a time value to `wdStartTime` and `wdEndTime`. It expects a time in the 24hour format as shown below.
+
+```gantt
+        title A Gantt Diagram
+        accTitle: A simple sample gantt diagram
+        accDescr: 2 sections with 2 tasks each, from 2014
+        dateFormat  YYYY-MM-DD
+        wdStartTime 08:00
+        wdEndTime 17:00
+```
```

Comment:
Can we use the YAML config for this, instead of adding more custom syntax?
````suggestion
You can assign core working hours within the Gantt by providing a time value to `wdStartTime` and `wdEndTime`. It expects a time in the 24hour format as shown below.

```
---
title: A Gantt Diagram
config:
  gantt:
    dateFormat: YYYY-MM-DD
    workdayStartTime: 08:00
    workdayEndTime: 17:00
---
gantt
        accTitle: A simple sample gantt diagram
        accDescr: 2 sections with 2 tasks each, from 2014
```
````

Reply:
Sorry can you explain in a bit more detail what you mean? Does this relate to not using jison?

Reply:
Yes, it was to move new configurations into the yaml config section, and not to add more keywords in jison.
I only edited the documentation, to reflect how it'll look after the change.

Reply:
@nirname  
The below is created within the config.type.ts automatically by the YAML config file (wdStartTime and wdEndTime are the two new pieces of code. However, I want to import this into the GanttDB.js file but as i understand it 'interface' is a typescript definition that cannot be imported in a plain JS file. What methods are there for getting around this? Or is it purely a need to re-write the GanttDB file? 

```
export interface GanttDiagramConfig extends BaseDiagramConfig {
  /**
   * Margin top for the text over the diagram
   */
  titleTopMargin?: number;
  /**
   * The height of the bars in the graph
   */
  barHeight?: number;
  /**
   * The margin between the different activities in the gantt diagram
   */
  barGap?: number;
  /**
   * Margin between title and gantt diagram and between axis and gantt diagram.
   *
   */
  topPadding?: number;
  /**
   * The space allocated for the section name to the right of the activities
   *
   */
  rightPadding?: number;
  /**
   * The space allocated for the section name to the left of the activities
   *
   */
  leftPadding?: number;
  /**
   * Vertical starting position of the grid lines
   */
  gridLineStartPadding?: number;
  /**
   * Font size
   */
  fontSize?: number;
  /**
   * Font size for sections
   */
  sectionFontSize?: string | number;
  /**
   * The number of alternating section styles
   */
  numberSectionStyles?: number;
  /**
   * Date/time format of the axis
   *
   * This might need adjustment to match your locale and preferences.
   *
   */
  axisFormat?: string;
  /**
   * axis ticks
   *
   * Pattern is:
   *
   * ```javascript
   * /^([1-9][0-9]*)(millisecond|second|minute|hour|day|week|month)$/
   * ```
   *
   */
  tickInterval?: string;
  /**
   * When this flag is set, date labels will be added to the top of the chart
   *
   */
  topAxis?: boolean;
  /**
   * Controls the display mode.
   *
   */
  displayMode?: '' | 'compact';
  /**
   * On which day a week-based interval should start
   *
   */
  weekday?: 'monday' | 'tuesday' | 'wednesday' | 'thursday' | 'friday' | 'saturday' | 'sunday';
  /**
   * Allows user to set the start time for the Workday. Used in calculating end times for tasks when set in hours
   *
   */
  wdStartTime?: string;
  /**
   * Allows user to set the end time for workday
   *
   */
  wdEndTime?: string;
```

Reply:
@sidharthv96 I'm thinking of a different approach which may not need yaml config.

We can extend includes/excludes properties to support time ranges.
Example:
```yaml
wdStartTime: 10:00
wdEndTime: 19:00
```
can be replaced by

```yaml
excludes: weekend, 19:00-10:00
```

Another benefit of this approach is that such a time range pattern can be added in "includes" also.

In later PRs, this idea can be extended to support DateTime to DateTime ranges also, enabling the user to define any working schedule only using includes/excludes patterns.